crates/hyperqueue/src/worker/start.rs

```diff
@@ -225,6 +229,47 @@ async fn run_task(
     Ok(TaskResult::Finished)
 }
 
+/// Provide a more detailed error message when a process fails to be spawned.
+fn map_spawn_error(error: std::io::Error, program: &ProgramDefinition) -> tako::Error {
+    let context = match &error.kind() {
+        ErrorKind::NotFound => {
+            let file = &program.args[0];
+            let mut msg = format!(
+                "\nThe program that you have tried to execute (`{}`) was not found.",
+                file
+            );
+
+            let path = Path::new(OsStr::from_bytes(file.as_bytes()));
+            if !path.is_absolute() && !path.starts_with(".") {
+                let possible_path = program.cwd.as_ref().unwrap().join(path);
+                if possible_path.exists() {
```

Comment:
Maybe ``is_file`` is better than ``exists``?

Reply:
Good point! We could also check that it's executable, but I'm already a bit afraid of FS latency here, since it's synchronous, so let's keep it at `is_file` for now.